app/assets/main/react_components/footprint_result/Payment.jsx

```diff
@@ -1,29 +1,206 @@
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { useLocaleData } from '../context/Footprint/LocaleContext';
 import { useTexts } from '../context/Footprint/TextsContext';
+import PriceText from './PriceText.jsx';
+
+/**
+ * Component for inputing payment details and doing a payment
+ */
+const Payment = ({
+  selectedMembership,
+  lifestyleFootprint,
+  grantedReferralCode,
+  multipleOffsets,
+  price,
+}) => {
+  const [userEmail, setUserEmail] = useState('');
+  const [userPassword, setUserPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [cardErrorMessage, setCardErrorMessage] = useState('');
+  const [loadingIconState, setLoadingIconState] = useState('hidden');
+
+  const mounted = useRef(false);
+
+  const Elements = useElements();
+  const stripe = useStripe();
 
-const Payment = ({ selectedMembership }) => {
   const {
     commonText: { email, password, credit_or_debit_card },
+    registrationsText: { start_subscription },
   } = useTexts();
+  const { currentRegion, slug } = useLocaleData();
+
+  const csrfToken = document.querySelector('meta[name="csrf-token"]').content;
+
+  const signUp = (paymentMethodID = undefined) => {
+    return fetch(slug + '/users.json', {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'X-CSRF-Token': csrfToken,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        lifestyle_footprint: lifestyleFootprint,
+        people: selectedMembership === 'multi' ? multipleOffsets : 1,
+        referral_code: grantedReferralCode,
+        membership: selectedMembership,
+        user: {
+          region: currentRegion,
+          email: userEmail,
+          password: userPassword,
+        },
+        payment_method_id: paymentMethodID,
+      }),
+    })
+      .then((response) => {
+        if (mounted.current) {
+          response.json().then((response) => {
+            switch (response.next_step) {
+              case 'redirect':
+                window.location = response.success_url;
+                break;
+              case 'confirmation_required':
+                confirm(
+                  response.intent_type,
+                  response.intent_client_secret
+                ).then((result) => {
+                  if (result.error !== undefined) {
+                    setErrorMessage(result.error.message);
+                  } else {
+                    window.location = response.success_url;
+                  }
+                });
+                break;
+              default:
+                setErrorMessage(response.error.message);
+                break;
+            }
+          });
+        }
+      })
+      .catch((error) => {
+        if (error.cardError) {
+          setErrorMessage(error.message);
+          return;
+        }
+        window.Sentry.captureException(error); // These errors are unexpected, so report them.
```

Comment:
Vad är det för kommentar på denna rad?

Reply:
Den fanns med i Rails också!

Reply:
Det är kopierat från deras JS-implementation, antog att det var error-hantering som de behövde, och de hade kommenterat så jag behöll det när jag la in!

Reply:
Den kommer från Rails!

Reply:
ok
👍😎